### PR TITLE
build(deps): require Golang 1.17 (for Mac M1 compatibility)

### PR DESCRIPTION
**Pull request by michaelfig**
## Description

Update Golang dependencies to 1.17 for compatibility with Mac M1 chip.

### Documentation Considerations

Should be updated on the Documentation site, too.

### CODE REVIEW

File: golang/cosmos/app/version_check.go [hunk 1]
```diff
@@ -1,6 +1,6 @@
-//+build !go1.15
+//+build !go1.17
 
 package gaia
 
-cosmic-swingset requires requires Golang version 1.15.
+cosmic-swingset requires requires Golang version 1.17.
```
Comment (dckc) [inline golang/cosmos/app/version_check.go]: This is golang syntax?!

File: packages/cosmic-swingset/README.md [hunk 2]
```diff
@@ -26,7 +26,7 @@ To build and install from sources, first follow the instructions at [Before Usin
 - **Warning:** There are some [known issues](https://github.com/Agoric/cosmic-swingset/issues/71) installing cosmic-swingset with [snap-based version of node.js on Ubuntu](https://github.com/nodesource/distributions/blob/master/README.md#snap). **We recommend** using a non-snap version
 
 You'll then need to install the following additional software:
-- [Golang](https://golang.org/doc/install) (you need at least version 1.15)
+- [Golang](https://golang.org/doc/install) (you need at least version 1.17)
```
Comment (dckc) [inline packages/cosmic-swingset/README.md:29]: :+1:

File: go.mod [hunk 3]
```diff
@@ -20,6 +20,104 @@ require (
 	gopkg.in/yaml.v2 v2.4.0
 )
 
+require (
+	filippo.io/edwards25519 v1.0.0-beta.2 // indirect
+	github.com/99designs/keyring v1.1.6 // indirect
+	github.com/ChainSafe/go-schnorrkel v0.0.0-20200405005733-88cbf1b4c40d // indirect
+	github.com/DataDog/zstd v1.4.5 // indirect
```
Comment (dckc) [inline go.mod:27]: This `require` thing is a consequence of going to 1.17?

Comment (dckc) [reply go.mod:27]: oic... `go mod tidy`

**Comment by dckc** [comment]
> ### Documentation Considerations
> Should be updated on the Documentation site, too.

To date, the documentation site is aimed at smart contract developers and we don't ask them to do anything that involves go.

Contributors to agoric-sdk need go; expectations for that audience conventionally belong in `CONTRIBUTING.md`. Would you please add a note there? Or should I do it stacked on top of this PR?

File: golang/cosmos/app/version_check.go (hunk 1 above)
Comment (michaelfig) [reply golang/cosmos/app/version_check.go]: It's syntactically-invalid stuff that should cause an error when not using 1.17, to point folks in the right direction before their build fails for other reasons.  But there are too many `requires requires`.

**Comment by michaelfig** [comment]
> To date, the documentation site is aimed at smart contract developers and we don't ask them to do anything that involves go.

Okay.

> Contributors to agoric-sdk need go; expectations for that audience conventionally belong in `CONTRIBUTING.md`. Would you please add a note there? Or should I do it stacked on top of this PR?

I'd appreciate it if you added a commit to this PR to do that.  I won't force-push anymore.  Stacked PR is fine, too.

**Comment by dckc** [comment]
> I'd appreciate it if you added a commit to this PR to do that.

OK: 280e95e

**Review by dckc — APPROVED** [review]

**Review by JimLarson — APPROVED** [review]

**Pull request merged**